packages/ubuntu_init/integration_test/e2e_test.dart

```diff
@@ -0,0 +1,113 @@
+import 'dart:io';
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:ubuntu_init/ubuntu_init.dart';
+import 'package:ubuntu_provision/ubuntu_provision.dart';
+import 'package:ubuntu_provision_test/ubuntu_provision_test.dart';
+import 'package:ubuntu_service/ubuntu_service.dart';
+import 'package:ubuntu_test/ubuntu_test.dart';
+import 'package:yaru_test/yaru_test.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  setUpAll(YaruTestWindow.ensureInitialized);
+
+  tearDown(() async {
+    await resetAllServices();
+    rootBundle.clear();
+  });
+
+  testWidgets('init wizard', (tester) async {
+    final windowClosed = YaruTestWindow.waitForClosed();
+
+    await tester.runApp(() => runInitApp([]));
+
+    await tester.testLocalePage(language: 'Deutsch');
+    await tester.tapNext();
+    await tester.pumpAndSettle();
+
+    await tester.testKeyboardPage(layout: 'Englisch (Britisch)');
+    await tester.tapNext();
+    await tester.pumpAndSettle();
+
+    await expectCommand('localectl', [], [
+      contains('LANG=de_DE.UTF-8'),
+      contains('X11 Layout: gb'),
+    ]);
+
+    // TODO: Test gsettings input-sources
+
+    await tester.testNetworkPage(mode: ConnectMode.none);
```

Comment:
Only test I can think of would be maybe to check the interface shown in the network page lines up with netplan? But otherwise I imagine this is difficult to test, especially since most the UI and interactions exist for wifi setups :sweat_smile:

Reply:
It's a bit cumbersome to deal with the network setup of a github runner. Also since networkmanager and the respective libraries are well-tested elsewhere, I don't think we need to test it extensively here. It's about testing the communication between the UI and provd after all.

Reply:
Fair enough :smile: